src/game.cpp

```diff
@@ -12717,9 +12717,9 @@ tripoint game::find_or_make_stairs( map &mp, const int z_after, bool &rope_ladde
     // Check the destination area for lava.
     if( mp.ter(stairs) == t_lava ) {
         if( movez < 0 &&
-            !query_yn(_("There is a LOT of heat coming out of there.  Descend anyway?")) ) {
+            !query_yn(_("There is a LOT of heat coming out of there.  Descend and enter the lava? You will not be able to get back up.")) ) {
```

Comment:
Note that the player doesn't know that it's lava yet (hence "There is a LOT of heat coming out of there")

Maybe: 

> There is a LOT of heat coming out of there, even the stairs have melted away.  Jump down? You won't be able to get back up.

This doesn't work as well for the ascent one, granted.